src/content/github/common/fetchers.ts

```diff
@@ -1,84 +1,78 @@
 import browser from "webextension-polyfill";
-import { MessageType } from "src/types";
+import {
+  FileCoverageReportResponse,
+  FileMetadata,
+  MessageType,
+} from "src/types";
+
+export async function getMetadata(url: string): Promise<FileMetadata> {
+  const response = await fetch(url).then((response) => response.json());
```

Comment:
For this to work with private repos do we need to pass `credentials: 'include'` to `fetch`? (https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API/Using_Fetch#sending_a_request_with_credentials_included)

Reply:
Works with the default value (same-origin)